lib/resources/resource_test.go

```diff
@@ -0,0 +1,112 @@
+package resources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+)
+
+// Tests that we can get Resource info from a Zabbix API call
+func TestGetInfo(t *testing.T) {
+	expected := Result{
+		{
+			HostID:      "10261",
+			Host:        "dummy-host",
+			Name:        "dummy-host",
+			Description: "",
+		},
+	}
+	// mock api call
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		res := struct {
+			Result `json:"result"`
+		}{expected}
+		err := json.NewEncoder(w).Encode(res)
+		if err != nil {
+			t.Fatalf("unable to write response. error: %v", err)
+		}
+	}))
+	defer ts.Close()
+
+	var z Zabbix
+	res, err := z.getInfo("ZABBIX_KEY", ts.URL)
+	if err != nil {
+		t.Fatalf("process ran with err %v, want result to be %v", err, expected)
+	}
+
+	if !(reflect.DeepEqual(res, expected)) {
+		t.Errorf("expected result to be %v", expected)
```

Comment:
Might be helpful to also print what was got.
```suggestion
		t.Errorf("expected result to be %v but got %v", expected, res)
```